newIDE/app/src/ObjectEditor/Editors/SpriteEditor/SpritesList.js

```diff
@@ -1,160 +1,239 @@
+    return (
+      <div style={styles.spritesList}>
+        {[
+          ...mapFor(0, spritesCount, i => {
+            const sprite = direction.getSprite(i);
+            return (
+              <SortableSpriteThumbnail
+                sprite={sprite}
+                key={sprite.ptr}
+                index={i}
+                selected={!!selectedSprites[sprite.ptr]}
+                onContextMenu={(x, y) => onSpriteContextMenu(x, y, sprite)}
+                onSelect={selected => onSelectSprite(sprite, selected)}
+                resourcesLoader={resourcesLoader}
+                project={project}
+              />
+            );
+          }),
+          <AddSpriteButton
+            displayHint={!direction.getSpritesCount()}
+            key="add-sprite-button"
+            disabled
+            index={spritesCount}
+            onAdd={onAddSprite}
+            onEdit={onEditSprites}
+          />,
+        ]}
+      </div>
+    );
+  }
+);
+
+export default class SpritesList extends Component {
+  componentDidMount() {
+    if(!ipcRenderer){return};
+    ipcRenderer.on('piskelSavedChanges', (event, piskelFramePaths) => {
+      const { direction, project } = editedAnimationProp.props;
+      const resourcesManager = project.getResourcesManager();
+      direction.removeAllSprites(); /// clear the old sprite list
+      var i = 0; /// ...We need to recreate it in order to account for any new/removal/reorder frame changes made in piskel
+      for (i = 0; i < piskelFramePaths.length; i++) {
+        var imagePath = piskelFramePaths[i];
+        const imageResource = new gd.ImageResource();
+        imageResource.setFile(imagePath);
+        imageResource.setName(imagePath);
+        resourcesManager.addResource(imageResource);
+        const sprite = new gd.Sprite();
+        sprite.setImageName(imageResource.getName());
+        direction.addSprite(sprite);
+        imageResource.delete();
+        sprite.delete();
+      }
+
+      editedAnimationProp.forceUpdate();
+    });
+  };
+
+  onSortEnd = ({ oldIndex, newIndex }) => {
+    this.props.direction.moveSprite(oldIndex, newIndex);
+    this.forceUpdate();
+  };
+
+  onAddSprite = () => {
+    const {
+      resourceSources,
+      onChooseResource,
+      project,
+      direction,
+    } = this.props;
+    if (!resourceSources) return;
+    const sources = resourceSources.filter(source => source.kind === 'image');
+    if (!sources.length) return;
+
+    onChooseResource(sources[0].name).then(resources => {
+      resources.forEach(resource => {
+        project.getResourcesManager().addResource(resource);
+
+        const sprite = new gd.Sprite();
+        sprite.setImageName(resource.getName());
+        direction.addSprite(sprite);
+      });
+
+      this.forceUpdate();
+    });
+  };
+
+  onEditSprites = () => {
+    if(!electron){
+      alert("This feature is only supported in the Desktop version for now!\nYou can Download it from Gdevelop's website...");

```

Comment:
For consistency with the rest of the code base, you can use `Window.showMessageBox(...)` (`import Window from  "../../../Utils/Window"`)

Reply:
done